external/llvm-project/mlir/lib/Dialect/Async/Transforms/AsyncToAsyncRuntime.cpp

```diff
@@ -607,6 +703,34 @@ class AssertOpLowering : public OpConversionPattern<cf::AssertOp> {
   llvm::DenseMap<func::FuncOp, CoroMachinery> &outlinedFunctions;
 };
 
+//===----------------------------------------------------------------------===//
```

Comment:
Why do we need this? Shouldn't `--canonicalize` do this?

Reply:
maybe..